### PR TITLE
4.06 support for GCC 8

### PR DESCRIPTION
**Pull request by christophe-riolo**
I have cherry-picked two commits from trunk that solve issues making it impossible to compile this branch.

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -7,6 +7,12 @@
   Windows again (fault introduced by 4.06 Unicode changes).
   (David Allsopp)
 
+### Runtime system:
+
+- GPR#1752: do not alias function arguments to sigprocmask (Anil Madhavapeddy)
+
+- GPR#1753: avoid potential off-by-one overflow in debugger socket path
+  length (Anil Madhavapeddy)
```
Comment (gasche) [inline Changes]: could you keep a blank line after this line?

**Review by gasche — APPROVED** [review]
This seems reasonable to me (provided the CI passes), but see the minor comment on the Changes entry.

File: Changes (hunk 1 above)
Comment (gasche) [reply Changes]: The spacing of the Changes entry is also arguably wrong, but it was also wrong in the original Changes entry. Would you mind moving the credit line to its own line in both cases?

Comment (gasche) [reply Changes]: (The general format of Changes entry is documented in [CONTRIBUTING.md#changelog](https://github.com/ocaml/ocaml/blob/trunk/CONTRIBUTING.md#changelog))

**Comment by pmetzger** [comment]
If there's going to be a 4.06.2, I'd recommend also picking up my documentation change about safe-string...

**Comment by gasche** [comment]
I'm not currently aware of any plans to make a 4.06.2 release, and doing a new minor release of not-the-last-minor-version has never happened in the past (although it could happen and 4.07 is not out yet). For now my assumption was that some people build from the 4.06 branch, or distribute downstream packages of 4.06 that need to backport those patches anyway, so having an up-to-date 4.06 branch is convenient for them.

**Comment by damiendoligez** [comment]
We did release 4.04.2 after 4.05.0+beta1, so there is a precedent. But I currently don't have any plans for a 4.06.2.

File: Changes (hunk 1 above)
Comment (christophe-riolo) [reply Changes]: Sure, I took that entry as it was but I can edit it. I'll do that when I'm back to my computer!

Comment (christophe-riolo) [reply Changes]: I also was unsure of whether to leave the original PR number, but I assumed it was the correct choice as the discussions took place there, do you confirm that choice? 
Edit: no I'll write an edited changelog and reference the original PR rather that sounds better I suppose.

Comment (gasche) [reply Changes]: Leaving the original PRs is fine.
(It makes it very easy for people comparing the Changelogs to understand which entries have been moved around after the fact.)

**Comment by christophe-riolo** [comment]
Changelog updated to fit the format :)

**Comment by gasche** [comment]
Merged. thanks!

**Pull request merged**